lib/task_helpers/imports/custom_buttons.rb

```diff
@@ -0,0 +1,78 @@
+require "English"
+
+module TaskHelpers
+  class Imports
+    class CustomButtons
+      class ImportArInstances
+        def self.import(obj_hash)
+          new.import(obj_hash)
+        end
+
+        def import(obj_hash)
+          ActiveRecord::Base.transaction { obj_hash.each { |obj_def| create_object(*obj_def) } }
+        end
+
+        def create_object(class_name, obj_array)
+          klass = class_name.camelize.constantize
+
+          obj_array.collect do |obj|
+            begin
+              klass.create!(obj['attributes'].except('guid')).tap do |new_obj|
+                if obj['children'].present?
+                  obj['children'].each do |child|
+                    new_obj.add_members(create_object(*child))
+                  end
+                end
+
+                if obj['associations'].present?
+                  obj['associations'].each do |hoo|
+                    new_obj.send("#{hoo.first}=", create_object(*hoo).first)
+                  end
+                end
+
+                if obj['userid'].present?
+                  check_user(obj)
```

Comment:
Should `check_user` be passed `new_obj`?

Since code complexity is high on this method maybe we should break the if-blocks out into separate methods.

```ruby
klass.create!(obj['attributes'].except('guid')).tap do |new_obj|
  add_children(obj, new_obj)
  add_associations(obj, new_obj)
  check_user(new_obj)
end
```